index.js

```diff
@@ -15,6 +15,15 @@ module.exports = function (res) {
 		unzip.statusMessage = res.statusMessage;
 		unzip.socket = res.socket;
 
+		unzip.once('error', function (err) {
+			if (err.code === 'Z_BUF_ERROR') {
+				res.emit('end');
+				return;
+			}
+
+			res.emit('error', err);
+		});
+
 		res.on('close', function () {
 			unzip.emit('close');
```

Comment:
Ugh, seems there's no official way: http://stackoverflow.com/questions/19277094/how-to-close-a-readable-stream-before-end Node.js streams really are the worst...